### PR TITLE
WIP - decouple data storage

### PR DESCRIPTION
**Pull request by oetyng**
WIP

### CODE REVIEW

File: sn_node/src/node/messaging/node_msgs.rs [hunk 1]
```diff
@@ -424,83 +421,44 @@ impl Node {
...
-                        &query,
-                        auth,
-                        origin,
-                        sender,
-                        #[cfg(feature = "traceroute")]
-                        traceroute,
-                    )
-                    .await,
-                ])
+                // Response is later sent back to the relaying elder
```
Comment (oetyng) [inline sn_node/src/node/messaging/node_msgs.rs:453]: *requesting elder

File: sn_node/src/node/messaging/update_section.rs [hunk 2]
```diff
@@ -20,76 +20,39 @@ use std::collections::BTreeSet;
...
-        })
+        Cmd::Data(cmd)
     }
 
-    /// Will send a list of currently known/owned data to relevant nodes.
+    /// If we are an adult, this will send a list of currently
+    /// known/owned data to relevant nodes.
     /// These nodes should send back anything missing (in batches).
+    ///
+    /// This below comment seems not yet implemented:
```
Comment (oetyng) [inline sn_node/src/node/messaging/update_section.rs:45]: Move to Q/issue

File: sn_interface/src/messaging/system/node_msgs.rs [hunk 3]
```diff
@@ -35,8 +35,8 @@ pub enum NodeCmd {
     },
     /// Tells an Adult to store a replica of the data
     ReplicateData(Vec<ReplicatedData>),
-    /// Tells an Adult to fetch and replicate data from the sender
-    SendAnyMissingRelevantData(Vec<DataAddress>),
+    /// Tells an Adult to send over data that the sender should have but doesn't.
+    ComplementMissingData(Vec<DataAddress>),
```
Comment (davidrusu) [inline sn_interface/src/messaging/system/node_msgs.rs:39]: The old name was more clear to me

File: sn_node/src/node/flow_ctrl/event_channel.rs [hunk 4]
```diff
@@ -6,27 +6,26 @@
...
     )
 }
 
-/// Sender for [`crate::node::Node`] events
+/// Sender for arbitrary events.
 #[derive(Clone)]
-pub(crate) struct EventSender {
-    event_sender: mpsc::Sender<Event>,
+pub(crate) struct EventSender<T> {
+    event_sender: mpsc::Sender<T>,
```
[the remark this answers is not in the document]
Comment (oetyng) [reply sn_node/src/node/flow_ctrl/event_channel.rs]: I avoided to touch them altogether in this PR.

**Comment by oetyng** [comment]
Superseded by #1691.

**Pull request closed without merging**